src/error.rs

```diff
@@ -0,0 +1,61 @@
+use std::error;
+use std::fmt;
+/// The error type returned by methods in this crate.
+pub enum Error<E> {
+    /// Manager Errors
+    Inner(E),
+    /// Timeout
+    Timeout,
+    /// BadConn
+    BadConn,
+    /// The pool has been closed or dropped
+    PoolClosed,
+}
+
+impl<E> From<E> for Error<E> {
+    fn from(e: E) -> Error<E> {
+        Error::Inner(e)
+    }
+}
+
+impl<E> fmt::Display for Error<E>
+where
+    E: fmt::Display + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Inner(ref err) => write!(f, "{}", err),
+            Error::Timeout => write!(f, "Timed out in mobc"),
+            Error::BadConn => write!(f, "Bad connection in mobc"),
+            Error::PoolClosed => write!(f, "The pool is closed"),
+        }
+    }
+}
+
+impl<E> fmt::Debug for Error<E>
```

Comment:
Do we really need this?

Reply:
Probably not. But I'm going to leave it there for now.

Reply:
Looking at it, you are correct. I'm going to delete.

Reply:
Ah, its needed for the `impl<E> error::Error for Error<E>` below.